api/types/constants.go

```diff
@@ -1282,9 +1282,15 @@ var KubernetesClusterWideResourceKinds = []string{
 
 const (
 	// TeleportServiceGroup is a default group that users of the
-	// teleport automated user provisioning system get added to so
-	// already existing users are not deleted
+	// teleport automated user provisioning system get added to
+	// when provisioned in INSECURE_DROP mode. This prevents
+	// already existing users from being tampered with or deleted.
 	TeleportServiceGroup = "teleport-system"
+	// TeleportManagedGroup is a default group that users of the
+	// teleport automated user provisioning system get added to
+	// when provisioned in KEEP mode. This prevents already
+	// existing users from being tampered with or deleted.
+	TeleportManagedGroup = "teleport-managed"
```

Comment:
Can we rename these constants so it's easier to tell them apart? (something like `TeleportDropGroup` and `TeleportKeepGroup`)

Reply:
yeah I think that makes a lot more sense, especially since we're adding `teleport-static` as well :+1:

For migrating `teleport-system`, I think we could just swap in `teleport-drop` for any newly created users and treat instances of `teleport-system` as if they were `teleport-drop`. I don't think it'd be too much extra effort and it's probably worth the saved future confusion